soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4059,6 +4181,46 @@ void DrawRandoEditor(bool& open) {
                     SohImGui::EnhancementCombobox("gRandomizeShuffleDungeonReward", randoShuffleDungeonRewards, 4, 0);
                     PaddedSeparator();
 
+                    // Keysanity
+                    ImGui::Text(Settings::Keysanity.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Own dungeon - Small Keys can only appear in their respective dungeon.\n"
+                        "\n"
+                        "Any dungeon - Small Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Small Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Small Keys can appear anywhere in the world."
+                    );
+                    SohImGui::EnhancementCombobox("gRandomizeKeysanity", randoShuffleSmallKeys, 6, 1);
+                    PaddedSeparator();
+
+                    // Gerudo Keys
+                    ImGui::Text(Settings::GerudoKeys.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Any dungeon - Small Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Small Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Small Keys can appear anywhere in the world."
+                    );
+                    SohImGui::EnhancementCombobox("gRandomizeGerudoKeys", randoShuffleGerudoFortressKeys, 4, 0);
+                    PaddedSeparator();
+
+                    // Boss Keysanity
+                    ImGui::Text(Settings::BossKeysanity.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Own dungeon - Boss Keys can only appear in their respective dungeon.\n"
+                        "\n"
+                        "Any dungeon - Boss Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Boss Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Boss Keys can appear anywhere in the world."
+                    );  
+                    SohImGui::EnhancementCombobox("gRandomizeBossKeysanity", randoShuffleBossKeys, 6, 1);
```

Comment:
hover text should have all 6, defaulting to `1` is vanilla

i think "Own Dungeon" would be good default here too